main/test/ca/uwaterloo/flix/library/TestMutList.flix

```diff
@@ -1730,4 +1662,28 @@ def sortBy!08(): Bool & Impure = testSortBy!VsSortBy(Array.toMutList([5,3,0,4,1,
 
 @test
 def sortBy!09(): Bool & Impure = testSortBy!VsSortBy(Array.toMutList([2,3,0,4,1,2]))
+
+    /////////////////////////////////////////////////////////////////////////////
+    // range                                                                   //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def range01(): Bool & Impure =
+        let v1 = MutList.range(1, 4) |> MutList.toArray;
+        let v2 = MutList.range(0, 5) |> MutList.toArray;
+        let arr1 = [1, 2, 3];
+        let arr2 = [0, 1, 2, 3, 4];
+        v1 == arr1 and v2 == arr2
+
+    @test
+    def range02(): Bool & Impure =
+        let v = MutList.range(-1, 3) |> MutList.toArray;
+        let arr = [-1, 0, 1, 2];
+        v == arr
+
+    @test
+    def range03(): Bool & Impure =
+        let MutList(a, l) = MutList.range(0, 4);
+        let arr = [0, 1, 2, 3, default, default, default, default];
```

Comment:
To me this feels like we're testing implementation more than necessary. What do you think?

Reply:
That was a sanity check I had to do when fiddling with arrays at the time. Refactored to normal test now. Thanks for pointing that out.